src/test/ui/parser/expr-as-stmt-2.stderr

```diff
@@ -36,6 +36,11 @@ LL | /     &&
 LL | |     if let Some(y) = a { true } else { false }
    | |______________________________________________^ expected `bool`, found `&&bool`
    |
+help: consider removing the `&&`
+   |
+LL -     &&
+LL +     if let Some(y) = a { true } else { false }
+   | 
```

Comment:
I think this suggestion's span is wrong. If you apply this suggestion, you'd end up with two copies of the `if let` expression.

Reply:
I think this span was already broken, see: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f2019826203ecd9a76dfb145b8517170

The issue here I think is [`demand.rs:843`](https://github.com/rust-lang/rust/pull/91545/files#diff-789a2e462c73d9f835b8c816c0cd8abca6d190e864a92f6a93d525c3b1dd62f7R843). We're using the `Span::lo` of the expression, which isn't exactly the `Span::hi` of the ampersand token.

Reply:
This actually might just be a bug in *rendering* help suggestions, since what it's trying to suggest here is removing `&&\n`. When I run `cargo fix --broken-code`, it seems to work fine.

Reply:
Hmm, interesting. I was wondering if it was a rendering issue. Can you open an issue about that?